src/Products/Products.js

```diff
@@ -0,0 +1,165 @@
+    showSideNav(true);
+  };
+
+  const handleClose = () => {
+    showSideNav(false);
+  };
+
+  return (
+    <div className='products'>
+      <div className='responsive-filter'>
+        <h3>Filters</h3>
+        <button className='resp-btn-filter' onClick={handleClick}>
+          <i className='fas fa-filter'></i>
+        </button>
+      </div>
+      <div className={sideNav ? "side-filters" : "main-filters"}>
+        <div className='resp-heading-filter'>
+          <h1>Apply Filters</h1>
+          <button className='closebtn' onClick={handleClose}>
+            <i className='fas fa-times'></i>
+          </button>
+        </div>
+        <Filters />
+      </div>
+      <div className='product-listing'>
+        <h2 className='center-txt ctn'>
+          {status.loading && (
+            <img src={Loading} alt='loading' className='loading' />
+          )}
+        </h2>
+        {rangedData.map((product) => {
+          return (
+            <div className='card' key={product._id}>
+              <img
+                className='card-img'
+                src={product.image}
+                alt={product.name}
+              />
+              <div className='bdge-sm pink'>
+                {product.inStock ? "Stock" : "Sold"}
+              </div>
+              <Link to={`/products/${product._id}`}>
+                <div className='card-info'>
+                  <p>{product.name}</p>
+
+                  <div className='card-info-details'>
+                    <h5>
+                      {product.material} - {product.brand}
+                    </h5>
+                    <div className='badge-ratings'>
+                      <h5>{product.ratings}</h5>
+                      <i className='fa fa-star'></i>
+                    </div>
+                  </div>
+                  <h5>
+                    {product.level} :
+                    {product.fastDelivery ? " Fast Delivery" : " 3 days +"}
+                  </h5>
+                  <h5>Category: {product.category}</h5>
+                  <div className='card-details'>
+                    <h5>Price: {product.price}</h5>
+                    <button
+                      onClick={
+                        token
+                          ? (e) => {
+                              e.preventDefault();
+                              return userState?.wishList.reduce(
+                                (acc, value) => {
+                                  return value._id === product._id
+                                    ? deleteFromWishListApi(
+                                        userState,
+                                        product,
+                                        userDispatch
+                                      )
+                                    : acc;
+                                },
+                                addToWishListApi(
+                                  userState,
+                                  product,
```

Comment:
Take this login outside the return statement in a separate function, and also avoid ternary operator if its not single line statement